src/WebJobs.Script/Environment/EnvironmentExtensions.cs

```diff
@@ -14,6 +15,14 @@ internal static class EnvironmentExtensions
         // For testing
         internal static string BaseDirectory { get; set; }
 
+        public static bool IsOutOfProc(this IEnvironment environment)
+        {
+            // TODO: need to also handle HttpWorker scenario?
+            // Is the env setting guaranteed to be set for OOP cases?
+            var runtime = environment.GetEnvironmentVariable(RpcWorkerConstants.FunctionWorkerRuntimeSettingName);
+            return string.IsNullOrEmpty(runtime) || !string.Equals(runtime, RpcWorkerConstants.DotNetLanguageWorkerName, StringComparison.OrdinalIgnoreCase);
```

Comment:
We do not enforce setting worker runtime env variable. It just takes precedence if set. Is it is ok to determine out-of-proc or not after JobHost starts? or do you need it info earlier? 
Reliable way to figure out out-of-proc would be look at function descriptor: https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Host/ScriptHost.cs#L510-L533